src/snowflake/cli/api/entities/common.py

```diff
@@ -1,15 +1,26 @@
 from enum import Enum
+from typing import Generic, TypeVar, get_args
 
 
 class EntityActions(str, Enum):
     BUNDLE = "bundle"
 
 
-class EntityBase:
+T = TypeVar("T")
+
+
+class EntityBase(Generic[T]):
     """
     Base class for the fully-featured entity classes.
     """
 
+    def __init__(self, entity_model: T):
+        self._entity_model = entity_model
+
+    @classmethod
+    def get_entity_model_type(cls) -> T:
+        return get_args(cls.__orig_bases__[0])[0]  # type: ignore[attr-defined]
```

Comment:
Can you add comment what's happening there?

Reply:
Done